File: Authinpl/UserRegistration.java
package main;

public class UserRegistration {
	
	public static User createUser(String username, String password) {
		User user=null;
		
		AuthHelper helper = new AuthHelper();
		String salt = helper.bytesToHex(helper.getSalt(32));
		String passsalt = password+salt;
		String hashedpass = helper.bytesToHex(helper.getHash(passsalt));
		
		user = new User(username, hashedpass, salt);
		
		return user;
	}

}

File: user_registration.py
from auth import get_salt, get_hash, bytes_to_hex, load_users, save_users, find_user
from user import User

def create_user(username, password):
    salt = bytes_to_hex(get_salt(32))
    pass_salt = password + salt
    hashed_pass = get_hash(pass_salt)
    return User(username, hashed_pass, salt)

def register_user(username, password):
    users = load_users('usersdb.json')
    if find_user(users, username):
        return False  # User already exists

    new_user = create_user(username, password)
    users.append(new_user)
    save_users(users, 'usersdb.json')
    return True
